src/main/java/com/gmail/yuramitryahin/service/TransactionServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.gmail.yuramitryahin.service;
+
+import com.gmail.yuramitryahin.entity.Account;
+import com.gmail.yuramitryahin.entity.Transaction;
+import com.gmail.yuramitryahin.repository.TransactionRepository;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@AllArgsConstructor
+@Service
+public class TransactionServiceImpl implements TransactionService {
+    private final TransactionRepository transactionRepository;
+    private final AccountService accountService;
+    private final CurrencyTranslationService currencyTranslation;
+
+    @Override
+    @Transactional
+    public void createTransfer(String fromAccountNumber, String toAccountNumber,
+                               BigDecimal amount) {
+        Account fromAccount = accountService.getByAccountNumber(fromAccountNumber);
+        Account toAccount = accountService.getByAccountNumber(toAccountNumber);
+        Transaction transactionFromAccount = Transaction.builder()
+                .fromAccount(fromAccount)
+                .toAccount(toAccount)
+                .dateTime(LocalDateTime.now())
+                .amount(amount)
+                .type(Transaction.TransactionType.OUTCOMING)
+                .build();
+        fromAccount.setBalance(fromAccount.getBalance().subtract(amount));
+        accountService.save(fromAccount);
+        transactionRepository.save(transactionFromAccount);
+
+        BigDecimal currencyAmount = currencyTranslation.convert(fromAccount.getCurrency()
```

Comment:
you will receive only exchange rate here - you should multiply amount on this rate

Reply:
nevermind, you can have the result amount as well if you pass the amount to the API :)